commons/FPath.ml

```diff
@@ -0,0 +1,28 @@
+
+(* This module governs a path type, which can only produce values which
+   are for normalized paths to existent files or directories.
+   These paths are always canonically kept at their absolute forms, in order to
+   maintain a canonicity property. Due to the `Path` constructor being private, we
+   can enforce this invariant at the type level.
```

Comment:
you can add a comment about the fact this file was originally called Path.ml but then it conflicts with files from the ocaml compiler libs